### PR TITLE
ES 159 Testing Changes

### PR DESCRIPTION
**Pull request by jaguillette**
These changes introduce functional fixes for the ROS/Gazebo app after testing with classroom instructions. The changes include:
- Parameterize ROS distribution so it's easier to update versions
- Add timestamps to different parts of the image creation process so that apptainer build logs are easier to parse
- Add ROS package dependency installs required by course
    - roslaunch
    - ur-gazebo
    - ur5-moveit-config
- Export needed environment variables during build process (e.g. update `PATH` with ROS install location).
- Install universal robots from GitHub, following install process from last year's Docker container
- Set required environment variables in the container definition so they're available when launched by a user
- Update container name
- Add a MATLAB installation with ROS interface software, so that students can use MATLAB to interface with ROS and Gazebo.

## MATLAB
The MATLAB software installation requires some setup. It needs
- Installation files
- An installation options file
- A network license file

The installation files require [downloading from MATLAB](https://www.mathworks.com/help/install/ug/download-without-installing.html), which requires downloading a MATLAB application, then using that to download the full installation files. Once the full installation files are downloaded, they need to be added to the shared EFS volume. So far this has been done by zipping the contents, transferring the zip file to S3, downloading it to the shared EFS, then decompressing the files. The end result should have the decompressed installation files at `/shared/MATLAB`, and `/shared/MATLAB/install` should be the location of the installer.

The installation options and the network license file are currently outside of version control, but not in a controlled, consistently accessible location. A critical future improvement to make this configuration more sustainable is to store these files in a consistent location in AWS. Ideally, this will be part of creating a scripted setup for these installation files to at least pull them all from S3.

These changes allow the first lesson instructions for the course to run, including with MATLAB as an interface. The python executable must be set in MATLAB when `rosinit` is first run. When running the command, it will first error out, complaining about Python. At the end of the error message, there is a link to change the python path. Changing this to match the output of `which python3` allows `rosinit` to complete its setup process. If this setting change can be baked into the image, that would be ideal.

## Homedir issue

This PR also includes a fix for an issue that was preventing the remote desktop from launching at all in the stage environment. The interactive app failed to launch, with the following message in the logs:

```
Unable to init server: Could not connect: Connection refused
xfce4-session: Cannot open display: .
```

The error seems to have been coming from the `xfce4-session` at the end of the `xfce.sh` script, running inside the apptainer container. I was having trouble finding the root cause, and had already transferred the same container built for the dev environment to the stage environment, and also set the SMVP image back to match dev. I even checked the checksums of the image files against each other to see if they were the same, but they were. Since the command was running inside the container, and the containers are identical, it's got to be something from outside the container somehow. I did a `printenv` on the (working) dev and (broken) stage app startup scripts, and found that the `HOME` variable was set to `/home/jeg390` instead of `/shared/home/jeg390` on the stage app, which is not good. Adding the line that's now in the script.sh.erb file that sets `APPTAINER_HOME=$HOME` resolved the issue, by explicitly passing the fixed home environment variable into the container. I'm still not sure how they got to be different values, but what's important for now is that it's fine.

See [Apptainer environment variable appendix](https://apptainer.org/docs/user/main/appendix.html) for details.

### CODE REVIEW

**Comment by jaguillette** [comment]
Note that I just added a fix for a problem that came up in moving the setup to stage to the end of the description for the PR.

File: template/script.sh.erb [hunk 1]
```diff
@@ -6,6 +6,7 @@ echo "TIMING: $(date -Iseconds) - Start script"
 echo "original HOME=$HOME"
 echo "user=$(id -nu)"
 export HOME="/shared/home/$(id -nu)"
+export APPTAINER_HOME=$HOME
```
Comment (Tanvez) [inline template/script.sh.erb:9]: Awesome find! @jaguillette Thanks for the detailed write-up, good to see your process in debugging this thorny detail.

**Review by Tanvez — COMMENTED** [review]
Thanks for the detailed write-up @jaguillette ! Everything looks great! We will probably need a run-book/docs update to outline/detail the installation process for remote desktop things.

**Review by Tanvez — APPROVED** [review]

**Review by arthurian — APPROVED** [review]
Awesome work @jaguillette! We reviewed this synchronously on 8/14 :+1:

**Pull request merged**